recipe/cross_compile_support.sh

```diff
@@ -15,19 +22,79 @@ if [[ "${HOST_PLATFORM}" != "${BUILD_PLATFORM}" ]]; then
             echo "CMAKE_CROSSCOMPILING_EMULATOR: " >> ${CI_SUPPORT}/${CONFIG}.yaml
-            echo "- /usr/bin/qemu-$(echo $HOST_PLATFORM | cut -b 7-)-static"  >> ${CI_SUPPORT}/${CONFIG}.yaml
+            echo "- /usr/bin/qemu-${HOST_PLATFORM_ARCH}-static"  >> ${CI_SUPPORT}/${CONFIG}.yaml
             echo "CROSSCOMPILING_EMULATOR: " >> ${CI_SUPPORT}/${CONFIG}.yaml
-            echo "- /usr/bin/qemu-$(echo $HOST_PLATFORM | cut -b 7-)-static"  >> ${CI_SUPPORT}/${CONFIG}.yaml
+            echo "- /usr/bin/qemu-${HOST_PLATFORM_ARCH}-static"  >> ${CI_SUPPORT}/${CONFIG}.yaml
+        fi
+
+
+        if [[ "${CUDA_COMPILER_VERSION}" == "11.2" ]] && [[ "${CDT_NAME}" == "cos8" || "${CDT_NAME}" == "cos7" ]]; then
+            # We use cdt_name=cos7 for rhel8 based nvcc till we figure out
+            # a stable cos8 replacement.
+            EXTRACT_DIR=$(mktemp -d)
+            pushd ${EXTRACT_DIR}
+                if [[ "${HOST_PLATFORM_ARCH}" == "aarch64" ]]; then
+                    CUDA_HOST_PLATFORM_ARCH=sbsa
+                else
+                    CUDA_HOST_PLATFORM_ARCH=${HOST_PLATFORM_ARCH}
+                fi
+                # download manifest for latest 11.2.x patch version
+                curl -L https://developer.download.nvidia.com/compute/cuda/repos/rhel8/${CUDA_HOST_PLATFORM_ARCH}/version_11.2.2.json > versions.json
+                # packages for which we also need to install the -devel version
+                # (names need "_" not "-" to match spelling in manifest)
+                declare -a DEVELS=(
+                    "cuda_cudart"
+                    "cuda_driver"
+                    "cuda_nvml"
+                    "cuda_nvrtc"
+                    "libcublas"
+                    "libcufft"
+                    "libcurand"
+                    "libcusolver"
+                    "libcusparse"
+                    "libnpp"
+                    "libnvjpeg"
+                )
+                # add xxx-devel package to manifest with same version (and formatting) as xxx
+                for pkg in "${DEVELS[@]}"; do
+                    DEVELQUERY="${DEVELQUERY:+${DEVELQUERY} |}} . += {\"${pkg}_devel\": {\"version\": .${pkg}.version}}"
+                done
+                # sanity check
+                echo "jq-query: $DEVELQUERY"
+                # update versions.json with devel packages
+                jq '${DEVELQUERY}' versions.json > versions.json
+                # sanity check
+                cat versions.json
+                # read names & versions for necessary RPMs from manifest; download & install them
+                jq 'keys[] as $k | "\($k)-11-2-\(.[$k] | .version)-1"' versions.json | while read d; do
+                    # normalize "_" -> "-"; also adapt "_dev" -> "-devel" (for cuda_nvml_dev)
+                    fn="$(echo $d | sed 's/_/-/g' | sed 's/-dev/-devel/g').${HOST_PLATFORM_ARCH}.rpm"
+                    echo "Downloading & installing: $fn"
```

Comment:
I think we want to filter down to the `-dev` and `-devel` packages here. I think that's what you're doing here, but my bash reading skills are failing me at the moment.

The runtime packages will generally be non-functional without a GPU anyway.

Reply:
To get an idea what I'm doing here, copy [payload](https://developer.download.nvidia.com/compute/cuda/repos/rhel8/sbsa/version_11.2.2.json) into https://jqplay.org/ and then add `keys[] as $k | "\($k)-11-2-\(.[$k] | .version)-1"` as the filter. The RHS then gives you the list of things I'm looping over.

To get from the contents of the manifest to the rpm-names, I need to replace "_" with "-" (in addition to the string interpolation in the jq query), and I'm replacing the `cuda_nvml_dev` with `cuda-nvml-devel`.

That output in the playground is without the extra devel-files being generated, but you can get an idea of what it's trying to do by replacing the filter with

```
. += {"cuda_devel": {"version": .cuda.version}} | keys_unsorted[] as $k | "\($k)-11-2-\(.[$k] | .version)-1"
```

Reply:
> The RHS then gives you the list of things I'm looping over.

For convenience (without the sed-ing & the whole devel shenanigans), the jq query based on the upstreams versions.json looks like:
```
"cuda-11-2-11.2.2-1"
"cuda_cudart-11-2-11.2.152-1"
"cuda_cuobjdump-11-2-11.2.152-1"
"cuda_cupti-11-2-11.2.152-1"
"cuda_cuxxfilt-11-2-11.2.152-1"
"cuda_gdb-11-2-11.2.152-1"
"cuda_nvcc-11-2-11.2.152-1"
"cuda_nvdisasm-11-2-11.2.152-1"
"cuda_nvml_dev-11-2-11.2.152-1"
"cuda_nvprof-11-2-11.2.152-1"
"cuda_nvprune-11-2-11.2.152-1"
"cuda_nvrtc-11-2-11.2.152-1"
"cuda_nvtx-11-2-11.2.152-1"
"cuda_samples-11-2-11.2.152-1"
"cuda_sanitizer_api-11-2-11.2.152-1"
"libcublas-11-2-11.4.1.1043-1"
"libcufft-11-2-10.4.1.152-1"
"libcurand-11-2-10.2.3.152-1"
"libcusolver-11-2-11.1.0.152-1"
"libcusparse-11-2-11.4.1.1152-1"
"libnpp-11-2-11.3.2.152-1"
"libnvjpeg-11-2-11.4.0.152-1"
"nsight_compute-11-2-2020.3.1.4-1"
"nsight_systems-11-2-2020.4.3.7-1"
"nvidia_driver-11-2-460.32.03-1"
```

Reply:
... which now makes me realise that 

```
"nsight_compute-11-2-2020.3.1.4-1"
"nsight_systems-11-2-2020.4.3.7-1"
"nvidia_driver-11-2-460.32.03-1"
```

are wrong. Which is annoying, because `nsight-compute-2020.3.1-2020.3.1.4-1.aarch64.rpm` is constructed from a different naming scheme than the libcu* packages (which always have "11-2" in there, even if the version is "11.4.x", c.f. `libcublas-11-2-11.4.1.1043-1.aarch64.rpm`)

Reply:
I was really hoping we could avoid needing the actual runtime libraries and could get away with just getting the stub libraries but it looks like things aren't packaged in a way to enable that :cry:

Thanks for the explanation. That looks good!